platform/ui/src/components/studyBrowser/StudyBrowser.js

```diff
@@ -1,62 +1,86 @@
-      return study.thumbnails.map((thumb, thumbIndex) => {
-        if (this.props.supportsDragAndDrop) {
-          return (
-            <ThumbnailEntryDragSource
-              key={`${studyIndex}_${thumbIndex}`}
-              {...study}
-              {...thumb}
-              id={`${studyIndex}_${thumbIndex}`}
-              onClick={this.props.onThumbnailClick}
-              onDoubleClick={this.props.onThumbnailDoubleClick}
-            />
-          );
-        } else {
-          return (
-            <div className="ThumbnailEntryContainer">
-              <ThumbnailEntry
-                key={`${studyIndex}_${thumbIndex}`}
-                {...study}
-                {...thumb}
-                id={`${studyIndex}_${thumbIndex}`}
-                onClick={this.props.onThumbnailClick}
-                onDoubleClick={this.props.onThumbnailDoubleClick}
-              />
-            </div>
-          );
-        }
-      });
-    });
-
-    const components = thumbnails.flat();
-    return (
-      <div className="StudyBrowser">
-        <div className="scrollable-study-thumbnails">{components}</div>
+              return (
+                <Thumbnail
+                  key={`${studyIndex}_${thumbIndex}`}
+                  id={`${studyIndex}_${thumbIndex}`} // Unused?
+                  // Study
+                  studyInstanceUid={studyInstanceUid} // used by drop
+                  // Thumb
+                  altImageText={altImageText}
+                  imageId={imageId}
+                  instanceNumber={instanceNumber}
+                  displaySetInstanceUid={displaySetInstanceUid} // used by drop
+                  numImageFrames={numImageFrames}
+                  seriesDescription={seriesDescription}
+                  seriesNumber={seriesNumber}
+                  stackPercentComplete={stackPercentComplete}
+                  // Events
+                  onClick={onThumbnailClick}
+                  onDoubleClick={onThumbnailDoubleClick}
+                />
+              );
+            });
+          })
+          .flat()}
       </div>
-    );
-  }
+    </div>
+  );
 }
 
+const noop = () => {};
+
+StudyBrowser.propTypes = {
+  studies: PropTypes.arrayOf(
+    PropTypes.shape({
+      studyInstanceUid: PropTypes.string.isRequired,
+      thumbnails: PropTypes.arrayOf(
+        PropTypes.shape({
+          altImageText: PropTypes.string,
+          displaySetInstanceUid: PropTypes.string.isRequired,
+          imageId: PropTypes.string,
+          instanceNumber: PropTypes.number,
+          numImageFrames: PropTypes.number,
+          seriesDescription: PropTypes.string,
+          seriesNumber: PropTypes.number,
+          stackPercentComplete: PropTypes.number,
+        })
+      ),
+    })
+  ).isRequired,
+  onThumbnailClick: PropTypes.func,
+  onThumbnailDoubleClick: PropTypes.func,
+};
+
+StudyBrowser.defaultProps = {
+  studies: [],
+  onThumbnailClick: noop,
```

Comment:
What is the point of the noop if the prop is not required?

Reply:
https://github.com/OHIF/Viewers/pull/946/files/097af230eed60c98f43ca5269a9fab283bd79199#diff-d586a02af305f42f8e214dd9e355cf93R44

It makes the binding prettier. Spreading props to set them for a component is a code smell, and this allows us to avoid some conditional logic.

**Previous:**
```
<Thumbnail
  key={`${studyIndex}_${thumbIndex}`}
  {...this.props}
/>
```

**Without Spread:**
```
<Thumbnail
  key={`${studyIndex}_${thumbIndex}`}
  onClick={e => {
    if (this.props.onThumbnailClick) {
      this.props.onThumbnailClick(e);
    }
  }
  onDoubleClick={e => {
    if (this.props.onThumbnailDoubleClick) {
      this.props.onThumbnailDoubleClick(e);
    }
  }
/>
```

**Without Spread + Noop:**
```
const noop = () => {}; // no operation

<Thumbnail
  key={`${studyIndex}_${thumbIndex}`}
  onClick={onThumbnailClick}
  onDoubleClick={onThumbnailDoubleClick}
/>
```

FWIW, this is a pattern used in the React codebase. Lodash even has a `noop` function, haha.

Reply:
That makes a lot of sense!